src/ApiTemplates/BoredAPICaller.js

```diff
@@ -1,37 +1,29 @@
 import axios from "axios";
-import React, { useEffect } from "react";
-
-function BoredAPICaller(props) {
+// Javascript function instead of a component as API caller should be functions and should be called with useEffect
+// Also when returning an empty <div>, it is an indication to convert to a function instead of component
+// eslint-disable-next-line import/no-anonymous-default-export
+export default async (numberOfParticipants) => {
   const expressServerBoredApiUrl = "http://localhost:5000/boredApi";
 
-  useEffect(() => {
-    (async () => {
-      try {
-        const boredApiData = await axios.get(expressServerBoredApiUrl, {
-          params: {
-            participants: props.numberOfParticipants,
-          },
-        });
-        console.log(
-          boredApiData.data.activity +
-            boredApiData.data.type +
-            boredApiData.data.participants
-        );
-        props.userRecommendationsFromApi(
-          boredApiData.data.activity,
-          boredApiData.data.type
-        );
-      } catch (error) {
-        console.log(error.message);
-      }
-    })();
-  }, [props.isSubmitPressed]); // Removing the array doesn't work as no element here will
-  // undergo re-rendering as the useEffect would never be called. Hence
-  // the props.attribute is used to fetch API when those changes
-  return <div />;
-}
-export default BoredAPICaller;
-
+  try {
+    const boredApiData = await axios.get(expressServerBoredApiUrl, {
+      params: {
+        participants: numberOfParticipants,
+      },
+    });
+    console.log(
```

Comment:
quick note on console.log - if you want to log multiple things, you can pass them as many parameters, so no need to concatenate to one string

```js
console.log(boredApiData.data.activity, boredApiData.data.type, boredApiData.data.participants)
```